### ISSUE TITLE
about IoU loss

### ISSUE THREAD

**Issue by Monoclinic** [megvii-research/PETR#89]
Hello, thanks for your great work.
I noticed that you have tried GIoU loss (in the code), however it is commented out and not used in the paper.
I wonder whether GIoU/3D-IoU loss is effective for transformer-based 3D detection? I referred to some paper and it seems that L1 is more popular.

**Comment by yingfei1016** [comment]
Hi，
The loss weight of GIoU loss is set 0.0，so GIoU is not used in the code. IoU loss is difficult to apply in **camera-based** methods because of the depth estimation problem (IoU overlap is small and sometimes no overlap).  In lidar-based methods, transfusion has used the IoU loss in assignment https://github.com/XuyangBai/TransFusion/blob/master/configs/transfusion_nusc_voxel_L.py#L225.

**Comment by Monoclinic** [comment]
> Hi， The loss weight of GIoU loss is set 0.0，so GIoU is not used in the code. IoU loss is difficult to apply in **camera-based** methods because of the depth estimation problem (IoU overlap is small and sometimes no overlap). In lidar-based methods, transfusion has used the IoU loss in assignment https://github.com/XuyangBai/TransFusion/blob/master/configs/transfusion_nusc_voxel_L.py#L225.

Thanks for your reply!
I recognized that some lidar-based and multimodal methods such as transfusion do applied IoU loss for assignment, however the final loss for training is L1 (so the assignment cost and training loss is not the same).
In fact I have tried some IoU-based methods in transfusion, but the result are even worse than pure L1. I'm now confused about the effectiveness of IoU loss in transformer models, and it is rarely used in recent transformer-based methods.
I would really appreciate it if you can give some advice or share some experience. Maybe I should adjust the ratio between diifferent losses and keep the final loss in alignment with the assignment cost?

**Comment by yingfei1016** [comment]
Hi,
I think IoU-based loss is hard to work in camera-based (not a special case of transformer) 3D detection. There are several main reasons:
(1) 3D IoU and BEV IoU are stricter than 2D IoU . Since the depth estimation is inaccurate, there may be no overlap between the prediction box and GT, and there is difficult to optimize it at the early stage of training. 
(2) The main problem of 3D is depth estimation (coordinates of the center), not the size of the object. In 3D space, the same objects (all cars) have similar size, and the network is easy to learn.  To some extent, IoU loss optimizes the shape of objects in 2D detection, thus improving the performance (the metric is based IOU). However the mAP metric in nuscenes only consider the center position.  

I suggest that you can learn from the IoU branch instead of IoU loss in 2D detection.  For example, design a center branch to score the location performance of the center.

**Comment by Monoclinic** [comment]
Thanks for your time and insightful advices. It helps me a lot.
